lib/constructs/interface.js

```diff
@@ -1463,7 +1463,47 @@ class Interface {
     const { idl, name } = this;
 
     this.str += `
-      exports.install = (globalObject, globalName) => {
+      const exposed = new Set([
+    `;
+
+    for (const globalName of this.exposed) {
+      this.str += `"${globalName}",\n`;
+    }
+
+    this.str += `
+      ]);
+
+      exports.install = (globalObject, globalNames) => {
+        let isExposed = false;
+    `;
+
+    if (this.legacyWindowAliases) {
+      this.str += "let isWindow = false;\n";
+    }
+
+    this.str += `
+        for (const globalName of globalNames) {
+          if (exposed.has(globalName)) {
+            isExposed = true;
+    `;
+
+    if (this.legacyWindowAliases) {
+      this.str += `
+            if (globalName === "Window") {
+              isWindow = true;
+            }
```

Comment:
Same comments as above. For the `isWindow` check, another `globalNames.some()` should be fine too.